### ISSUE TITLE
Extract .basis from ktx2.

### ISSUE THREAD

**Issue by fire** [BinomialLLC/basis_universal#318]
We can compress images into ktx2.

```
basisu::basis_compressor c;
// ...
const basisu::uint8_vec &buvec = c.get_output_ktx2_file();
```

How do we take a ktx2 and get a basis file with all the mipmaps, layers, and supporting both etc1s and uastc?

**Comment by fire** [comment]
Working in  the https://github.com/V-Sekai/godot/tree/gltfpack-basisu branch.

```
// https://github.com/V-Sekai/godot/commit/3adfbf559adddf5d5a991ea7a2c03a97f6cc3e65
// This takes a image and outputs a ktx2.
Vector<uint8_t> (*Image::basis_universal_ktx2_packer_ptr)(const Ref<Image> &, Image::UsedChannels) = nullptr;
// This takes a image and outputs an image which is internally a .basis.
Ref<Resource> (*Image::basis_universal_ktx2_unpacker_ptr)(const uint8_t *, int) = nullptr;
```

**Comment by richgel999** [comment]
>How do we take a ktx2 and get a basis file with all the mipmaps, layers, and supporting both etc1s and uastc?
Is the KTX2 uncompressed? That should be relatively easy, although we would need to write a KTX2 reader. (Which isn't hard, but any parser involves security concerns.)

**Comment by fire** [comment]
Is a KTX2 reader something that fits this library for purpose of png -> ktx2, and ktx2 -> basis -> bc7.

**Comment by MarkCallow** [comment]
There is already a widely supported `KHR_texture_basisu` extension for glTF which uses KTX v2 format files. It seems like https://github.com/V-Sekai/godot/tree/gltfpack-basisu is out-of-date or incorrect.

**Comment by fire** [comment]
Yes, its incorrect, thats the whole point of this bug. I can either implement support or close.

**Comment by MarkCallow** [comment]
Or you can use a non-broken gltf pack tool.
